apluslms_roman/backends/__init__.py

```diff
@@ -38,17 +38,22 @@ def from_config(cls, index, data):
                 raise RuntimeError("Missing image name (img) in step configuration: {}".format(data))
             return cls(
                 index,
-                clean_image_name(data['img']),
+                data.get('img'),
                 data.get('cmd'),
                 data.get('mnt'),
-                dict(data.get('env', {})),
+                data.get('env'),
                 data.get('name'),
             )
         else:
             return cls(index, clean_image_name(data))
 
     def __init__(self, ref, img, cmd=None, mnt=None, env=None, name=None):
-        self.img, self.cmd, self.mnt, self.env, self.name, self.ref = img, cmd, mnt, env, name, ref
+        self.img = clean_image_name(img)
+        self.env = dict(env) if env else {}
+        self.cmd = cmd if isinstance(cmd, str) else tuple(cmd)
+        self.mnt = mnt
+        self.name = name
+        self.ref = ref
```

Comment:
I do prefer sorting these in the same order as function arguments, when that doesn't make code hard to read.